autoscheduler/frontend/src/redux/reducers/schedules.ts

```diff
@@ -22,18 +24,77 @@ export interface ReplaceSchedulesAction {
     type: 'REPLACE_SCHEDULES';
     schedules: Meeting[][];
 }
-export type ScheduleAction = AddScheduleAction | RemoveScheduleAction | ReplaceSchedulesAction;
+export interface SaveScheduleAction {
+    type: 'SAVE_SCHEDULE';
+    index: number;
+}
+export interface UnsaveScheduleAction {
+    type: 'UNSAVE_SCHEDULE';
+    index: number;
+}
+export type ScheduleAction = AddScheduleAction | RemoveScheduleAction
+  | ReplaceSchedulesAction | SaveScheduleAction | UnsaveScheduleAction;
+
+interface Schedules {
+  schedules: Meeting[][];
+  savedSchedules: Meeting[][];
```

Comment:
Doesn't this result in duplication of data? I'm thinking `savedSchedules` should be like an array of indices

Reply:
No, objects in arrays are references to the objects, not the actual data so storing it as a number or a reference take up the same memory, as both are 8 bytes